### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by dariia-romanova**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   "license": "GPL-3.0",
   "devDependencies": {
     "@mate-academy/eslint-config": "*",
-    "@mate-academy/scripts": "^0.3.5",
+    "@mate-academy/scripts": "^0.9.7",
```
Comment (WorkAccCom) [inline package.json:19]: Please do not add useless files to your PR.
You can use **git add src/** command to add needed files
**git status** command — for checking file status

**Review by WorkAccCom — APPROVED** [review]
LGTM 🔥

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,19 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```
Comment (AlinaMatuschak) [inline src/transformState.js:9]: try use switch case when you want to check a value for multiple possible results

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]
Almost perfect 👍

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
```
Comment (AlinaMatuschak) [inline src/transformState.js:8]:
```suggestion
  for (const action of actions) {
    const {type, extraData, keysToRemove} = action;
   
```

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,25 @@
...
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
```
Comment (AlinaMatuschak) [inline src/transformState.js:24]: it's last case, you can delete break

**Review by AlinaMatuschak — APPROVED** [review]
👌

**Review by AlinaMatuschak — APPROVED** [review]

**Pull request closed without merging**